tests/test_aggregates.py

```diff
@@ -1,362 +1,375 @@
-        with pytest.raises(LookupError):
-            aggregator_instance[get_aggregate_id((jobs[0],))]
-        # Test for an aggregate of all jobs for default aggregator
-        with pytest.raises(LookupError):
-            default_aggregator[get_aggregate_id(jobs)]
-
-    def test_contains(self, setUp, project):
-        jobs = tuple(project)
-        aggregator_instance = aggregator()._create_AggregateStore(project)
-        default_aggregator = aggregator.groupsof(1)._create_AggregateStore(project)
-        # Test for an aggregate of all jobs
-        assert get_aggregate_id(jobs) in aggregator_instance
-        assert get_aggregate_id(jobs) not in default_aggregator
-        # Test for an aggregate of single job
-        assert not jobs[0].id in aggregator_instance
-        assert jobs[0].id in default_aggregator
+from functools import partial
+from math import ceil
+
+import pytest
+import signac
+from conftest import FlowProjectSetup
+
+from flow.aggregates import _DefaultAggregateStore, aggregator, get_aggregate_id
+from flow.errors import FlowProjectDefinitionError
+
+
+class AggregateProjectSetup(FlowProjectSetup):
+    project_name = "AggregateTestProject"
+
+    def mock_project(self):
+        project = self.project_class.get_project(root=self._tmp_dir.name)
+        for i in range(10):
+            even = (i % 2) == 0
+            if even:
+                project.open_job(dict(i=i, half=i / 2, even=even)).init()
+            else:
+                project.open_job(dict(i=i, even=even)).init()
+        return project
+
+
+class AggregateFixtures:
+    @classmethod
+    def _get_single_job_aggregate(cls, jobs):
+        for job in jobs:
+            yield (job,)
+
+    @classmethod
+    def _get_all_job_aggregate(cls, jobs):
+        return (tuple(jobs),)
+
+    @pytest.fixture
+    def get_single_job_aggregate(self):
+        return AggregateFixtures._get_single_job_aggregate
+
+    @pytest.fixture
+    def get_all_job_aggregate(self):
+        return AggregateFixtures._get_all_job_aggregate
+
+    @classmethod
+    def list_of_aggregators(cls):
+        # The below list contains 14 distinct aggregator objects and some duplicates.
+        return [
+            aggregator(),
+            aggregator(),
+            aggregator(cls._get_all_job_aggregate),
+            aggregator(cls._get_single_job_aggregate),
+            aggregator(cls._get_single_job_aggregate),
+            aggregator.groupsof(1),
+            aggregator.groupsof(1),
+            aggregator.groupsof(2),
+            aggregator.groupsof(3),
+            aggregator.groupsof(4),
+            aggregator.groupby("even"),
+            aggregator.groupby("even"),
+            aggregator.groupby("half", -1),
+            aggregator.groupby("half", -1),
+            aggregator.groupby(["half", "even"], default=[-1, -1]),
+        ]
+
+    def create_aggregate_store(self, aggregator_instance, project):
+        return aggregator_instance._create_AggregateStore(project)

```

Comment:
I could call it directly, but it makes the code look more clean IMO. Happy to change this if you want

Reply:
The reason I'm stating this though is that if I remove this then I'll have to add multiple lines of code for initializing the aggregator_instance first. It might be painful to do that but if the current one is against best practices then I'm happy to change this.
@b-butler what do you suggest?